File: SmartShoppingAssistantLigaAc.DataAccess/Entities/Promotion.cs
using SmartShoppingAssistantLigaAc.DataAccess.Entities.Enums;

namespace SmartShoppingAssistantLigaAc.DataAccess.Entities;

public class Promotion
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public PromotionType Type { get; set; }
    public decimal Threshold { get; set; }
    public PromotionReward Reward { get; set; }
    public int RewardValue { get; set; }
    public int? ProductId { get; set; }
    public int? CategoryId { get; set; }
    public bool IsActive { get; set; }

    public Product? Product { get; set; }
    public Category? Category { get; set; }
}

File: Frontend/SmartShoppingAssistant.FE/src/components/shared/types/Promotion.ts
import type { PromotionModel } from "../../../api/models/PromotionModel"

export interface Promotion {
    id: number;
    name: string;
    type: string;
    typeLabel: string;        // Added back for the UI Table!
    reward: string;
    rewardLabel: string;      // Added back for the UI Table!
    threshold: number;
    rewardValue: number;
    productId: number | null;
    categoryId: number | null;
    isActive: boolean;
}

const typeLabels: Record<string, string> = {
    "Quantity": "Based on the quantity bought of a product",
    "CartTotal": "Based on the cart total"
};

const rewardLabels: Record<string, string> = {
    "FreeItems": "Buy X get Y free",
    "PercentDiscount": "Percentage discount"
};

export function toPromotion(dto: PromotionModel): Promotion {
    return {
        id: dto.id,
        name: dto.name,
        
        type: dto.type, 
        typeLabel: typeLabels[dto.type] ?? "Unknown",
        
        reward: dto.reward, 
        rewardLabel: `${dto.rewardValue} ${rewardLabels[dto.reward] ?? "Unknown"}`,
        
        threshold: dto.threshold,
        rewardValue: dto.rewardValue,
        productId: dto.productId,
        categoryId: dto.categoryId,
        isActive: dto.isActive
    };
}